### DISCUSSION TITLE
Typing actions

### DISCUSSION THREAD

**Discussion by dmwyatt** [statelyai/xstate#2045] [Q&A] [answered]
I've spent a couple of hours on this and just have no idea why I can't get this to type correctly.  I'm sure someone will look at this and point out something obvious I'm missing.

I think this is a minimal reproduction of the problem I'm having:

https://codesandbox.io/s/crazy-sunset-eblil?file=/src/index.ts

Note the error on line 58.  Anyone know what mistake I'm making?

**Comment by Andarist** [comment, marked as answer]
`DoneInvokeEvent` cannot currently be used as part of your `TEvent` union as it doesn't have a literal `type` property (it's typed as `string`) and thus our mapped types etc cannot use this union properly when you put it there.

**Reply by dmwyatt** [reply]
Oh!  Yeah, that does it. Thanks!

For future searchers coming across this discussion, here it is fixed: https://codesandbox.io/s/thirsty-newton-dgjry?file=/src/index.ts

Take `DoneInvokeEvent` out of your machine events and put it on your `assign`.
